logp/environment.go

```diff
@@ -47,7 +47,7 @@ const (
 	defaultEnvironmentString        = "default"
 	systemdEnvironmentString        = "systemd"
 	containerEnvironmentString      = "container"
-	macOSServiceEnvironmentString   = "macOS_service"
+	macOSServiceEnvironmentString   = "macos_service"
```

Comment:
This was copied from Beats, the original code is still used in 7.17:

https://github.com/elastic/beats/blob/9ded7059eb086d009d584266e83cf82306502516/libbeat/logp/environment.go#L46-L83

```go
// String returns the string representation the configured environment
func (v Environment) String() string {
	switch v {
	case DefaultEnvironment:
		return "default"
	case SystemdEnvironment:
		return "systemd"
	case ContainerEnvironment:
		return "container"
	case MacOSServiceEnvironment:
		return "macOS_service"
	case WindowsServiceEnvironment:
		return "windows_service"
	default:
		return "<invalid>"
	}
}

// ParseEnvironment returns the environment type by name.
// The parse is case insensitive.
// InvalidEnvironment is returned if the environment type is unknown.
func ParseEnvironment(in string) Environment {
	switch strings.ToLower(in) {
	case "default":
		return DefaultEnvironment
	case "systemd":
		return SystemdEnvironment
	case "container":
		return ContainerEnvironment
	case "macos_service":
		return MacOSServiceEnvironment
	case "windows_service":
		return WindowsServiceEnvironment
	default:
		return InvalidEnvironment
	}
}
```

It looks like we used both versions before and just arbitrarily picked one here. The important one seems to be the parse case which uses the lowercase spelling.

AFAIK nobody has complained about this as a bug so it is unlikely to be widely used, otherwise I'd suggest accepting both versions. Leaving it like this should be fine.

Reply:
This PR would still accept both versions (`macOS_service` and `macos_service`). 
The issue is that the current code (and APM Server) returns an error for both.

The current code expects `strings.ToLower(in) == "macOS_service"` which will never be true.

https://github.com/elastic/elastic-agent-libs/blob/fa8f242aaa3ef1292b29ee7761d7b9a083b4e04c/logp/environment.go#L76-L91

Reply:
Ah missed the ToLower conversion, thanks for clarifying. This indeed has never worked.